GA/problem_definition.py

```
@@ -0,0 +1,461 @@
+
+        # Validate critical submodels_path exist
+        if not self.submodels_path.exists():
+            raise SystemError(f"Missing submodels path: {self.submodels_path}")
+        
+        # Create folder path to store statefiles
+        self.dump_folder = self.submodels_path / "Evaluated_tdat_state_files"
+        # Check existance of dump folder
+        try:
+            self.dump_folder.mkdir(exist_ok=True)
+        except PermissionError:
+            raise PermissionError(f"Unable to create dump folder: {self.dump_folder}. Check permissions")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)